program-runtime/src/loaded_programs.rs

```diff
@@ -408,7 +412,9 @@ impl LoadedPrograms {
                 if let Some(second_level) = self.entries.get(&key) {
                     for entry in second_level.iter().rev() {
                         let current_slot = working_slot.current_slot();
-                        if current_slot == entry.deployment_slot
+                        if entry.deployment_slot <= self.latest_root
+                            || entry.deployment_slot == current_slot
+                            || entry.deployment_slot == 0
```

Comment:
`deployment_slot == 0`, why is that a special case ?

Reply:
On second thoughts, maybe it's not as useful. I've removed it in the latest commit.